### ISSUE TITLE
[suggestion] Add support for folia 

### ISSUE THREAD

**Issue by cgw2023** [mastercake10/TimeIsMoney#110]
### What steps will reproduce the issue?  
1. put plugin into plugins folder

### What was supposed to happen?
plugin to be load

### What happened instead? 
fail to load plugin

### What version of TimeIsMoney are you using?  
v1.9.6.22

### What Spigot version are you using? Paste the output of /version below.  
 git-Folia-"23b6f9e" (MC: 1.20)

### What plugins do you have installed? Paste the output of /plugins below.  
[17:03:36 INFO]: Server Plugins (4):
[17:03:36 INFO]: Bukkit Plugins:
[17:03:36 INFO]:  - AnnouncerPlus, Essentials, TimeIsMoney, Vault

### Are any errors related to TimeIsMoney in your console or logs? If so, paste below.  
```
[17:01:34 ERROR]: Error occurred while enabling TimeIsMoney v1.9.6.22 (Is it up to date?)
java.lang.UnsupportedOperationException: null
        at org.bukkit.craftbukkit.v1_20_R1.scheduler.CraftScheduler.handle(CraftScheduler.java:533) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at org.bukkit.craftbukkit.v1_20_R1.scheduler.CraftScheduler.runTaskTimer(CraftScheduler.java:233) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at org.bukkit.craftbukkit.v1_20_R1.scheduler.CraftScheduler.runTaskTimer(CraftScheduler.java:209) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at de.Linus122.TimeIsMoney.Main.startPlaytimeWatcher(Main.java:168) ~[TimeIsMoney-1.9.6.22.jar:?]
        at de.Linus122.TimeIsMoney.Main.onEnable(Main.java:138) ~[TimeIsMoney-1.9.6.22.jar:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:281) ~[folia-api-1.20.1-R0.1-SNAPSHOT.jar:?]
        at io.papermc.paper.plugin.manager.PaperPluginInstanceManager.enablePlugin(PaperPluginInstanceManager.java:189) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.enablePlugin(PaperPluginManagerImpl.java:104) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:507) ~[folia-api-1.20.1-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.v1_20_R1.CraftServer.enablePlugin(CraftServer.java:638) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at org.bukkit.craftbukkit.v1_20_R1.CraftServer.enablePlugins(CraftServer.java:549) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at net.minecraft.server.MinecraftServer.loadWorld0(MinecraftServer.java:679) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at net.minecraft.server.MinecraftServer.loadLevel(MinecraftServer.java:463) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:308) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1188) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:346) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
[17:01:34 INFO]: [TimeIsMoney] Disabling TimeIsMoney v1.9.6.22
[17:01:34 ERROR]: Error occurred (in the plugin loader) while disabling TimeIsMoney v1.9.6.22 (Is it up to date?)
java.lang.NullPointerException: Cannot invoke "org.bukkit.scheduler.BukkitTask.cancel()" because "this.playtimeWatcherTask" is null
        at de.Linus122.TimeIsMoney.Main.onDisable(Main.java:194) ~[TimeIsMoney-1.9.6.22.jar:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:283) ~[folia-api-1.20.1-R0.1-SNAPSHOT.jar:?]
        at io.papermc.paper.plugin.manager.PaperPluginInstanceManager.disablePlugin(PaperPluginInstanceManager.java:224) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.disablePlugin(PaperPluginManagerImpl.java:109) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at org.bukkit.plugin.SimplePluginManager.disablePlugin(SimplePluginManager.java:537) ~[folia-api-1.20.1-R0.1-SNAPSHOT.jar:?]
        at io.papermc.paper.plugin.manager.PaperPluginInstanceManager.enablePlugin(PaperPluginInstanceManager.java:193) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.enablePlugin(PaperPluginManagerImpl.java:104) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:507) ~[folia-api-1.20.1-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.v1_20_R1.CraftServer.enablePlugin(CraftServer.java:638) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at org.bukkit.craftbukkit.v1_20_R1.CraftServer.enablePlugins(CraftServer.java:549) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at net.minecraft.server.MinecraftServer.loadWorld0(MinecraftServer.java:679) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at net.minecraft.server.MinecraftServer.loadLevel(MinecraftServer.java:463) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:308) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1188) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:346) ~[folia-1.20.1.jar:git-Folia-"fa018cc"]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
```
### Any additional information that you would like to provide that may be relevant to the issue?

![image](https://github.com/mastercake10/TimeIsMoney/assets/136877036/184f3d74-6e3b-43c6-a0b2-48e9f8b1497d)

**Comment by mastercake10** [comment]
I have no idea how Folia works, seems to be a bit tricky to make the plugin work on Folia, Paper and Spigot at the same time. I don't understand why a simple scheduler wouldn't work on Folia since its just a global scheduler. Using their new `Bukkit#getAsyncScheduler` thingy (found it on their discord) requires the spigot api to be replaced with folia's api causing more conflicts in the rest of the code base.

I'll leave this open in case someone is familiar with Folia and is willing to solve this.

**Comment by mastercake10** [comment]
fixed in #113
